src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    const separate = `${separator}`;
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i] === null || this[i] === undefined
+        ? ''
+        : this[i];
+
+      i === this.length - 1
+        ? resultString += element
+        : resultString += element + separate;
```

Comment:
try to rewrite by using `if ... else` construction or use the ternary operator after setting the value to `resultString`: `resultString += i === this.length - 1 ? ... : ....;`